plugins/CoreHome/angularjs/common/services/periods.spec.js

```diff
@@ -0,0 +1,132 @@
+            piwikPeriods = $injector.get('piwikPeriods');
+        }));
+
+        it('should get daterange for day', function() {
+            var day = '2021-03-10';
+
+            var result = piwikPeriods.parse('day', day).getDateRange();
+            var expected = [clearDate(day), clearDate(day)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get daterange for week', function() {
+            var day = '2021-03-10';
+            var monday = '2021-03-08';
+            var sunday = '2021-03-14';
+
+            var result = piwikPeriods.parse('week', day).getDateRange();
+            var expected = [clearDate(monday), clearDate(sunday)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get daterange for month', function() {
+            var day = '2021-03-10';
+            var first = '2021-03-01';
+            var last = '2021-03-31';
+
+            var result = piwikPeriods.parse('month', day).getDateRange();
+            var expected = [clearDate(first), clearDate(last)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get daterange for month for date 31th', function() {
+            var day = '2021-03-31';
+            var first = '2021-03-01';
+            var last = '2021-03-31';
+
+            var result = piwikPeriods.parse('month', day).getDateRange();
+            var expected = [clearDate(first), clearDate(last)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get daterange for year', function() {
+            var day = '2021-03-10';
+            var first = '2021-01-01';
+            var last = '2021-12-31';
+
+            var result = piwikPeriods.parse('year', day).getDateRange();
+            var expected = [clearDate(first), clearDate(last)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get daterange for year for date 31th december', function() {
+            var day = '2021-12-31';
+            var first = '2021-01-01';
+            var last = '2021-12-31';
+
+            var result = piwikPeriods.parse('year', day).getDateRange();
+            var expected = [clearDate(first), clearDate(last)];
+
+            expect(result).to.eql(expected);
+        });
+
+        it('should get proper month rangeperiod when date is 31th march', function() {
+            var day = '2021-03-31';
+            var first = '2021-02-01';
+            var last = '2021-03-31';
+
+            var result = piwikPeriods.RangePeriod.getLastNRange('month', 2, day);
+
+            expect(result.startDate).to.eql(clearDate(first));
+            expect(result.endDate).to.eql(clearDate(last));
+        });
+
+        function mockDateNow01() {
+            return clearDate('2021-03-31').getTime();
+        }
+
+        it('should parse last month properly when date is 31th march', function() {
+            originalDateNow = Date.now;
+            Date.now = mockDateNow01;
```

Comment:
The additional method actually shouldn't be needed here and below. You could simply use a anonymous function

```suggestion
        it('should parse last month properly when date is 31th march', function() {
            originalDateNow = Date.now;
            Date.now = function() {
                return clearDate('2021-03-31').getTime();
            };
```

Reply:
Nice, I like it. Will make this change.